Source/Core/VideoCommon/XFStateManager.cpp

```diff
@@ -0,0 +1,279 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "VideoCommon/XFStateManager.h"
+
+#include "Common/ChunkFile.h"
+
+#include "VideoCommon/FramebufferManager.h"
+#include "VideoCommon/VertexManagerBase.h"
+#include "VideoCommon/XFMemory.h"
+
+void XFStateManager::Init()
+{
+  // Initialize state tracking variables
+  ResetTexMatrixAChange();
+  ResetTexMatrixBChange();
+  ResetPosNormalChange();
+  ResetProjection();
+  ResetViewportChange();
+  ResetTexMatrixInfoChange();
+  ResetLightingConfigChange();
+  ResetLightsChanged();
+  ResetMaterialChanges();
+  ResetPerVertexTransformMatrixChanges();
+  ResetPerVertexNormalMatrixChanges();
+  ResetPostTransformMatrixChanges();
+
+  std::memset(static_cast<void*>(&xfmem), 0, sizeof(xfmem));
```

Comment:
Seems like this could be:

```suggestion
  xfmem = {};
```

no?

Reply:
This was a move of the original source.  Isn't `Bitfield` not [directly copyable](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Common/BitField.h#L142)?  Maybe that is why `memset` was used?

Reply:
Ah, sorry, was thinking in terms of yuzu's variant of BitField which doesn't have the deleted copy constructor

Reply:
We really should refactor that one of these days, it's a bit of a pain at times...